apps/pipelines/nodes/base.py

```diff
@@ -41,7 +43,7 @@ def json_safe(self):
         return copy
 
     @classmethod
-    def from_node_output(cls, node_id: str, output: any = None, **kwargs) -> "PipelineNode":
+    def from_node_output(cls, node_id: str, output: any = None, **kwargs) -> Self:
```

Comment:
FYI, I was getting type hint errors from this change: 
```
Expression of type "PipelineState" cannot be assigned to return type "Self@PipelineState"
```
I'll update to return `cls(**kwargs)`